packages/react-server/src/ReactFizzServer.js

```diff
@@ -170,12 +183,18 @@ const ERRORED = 4;
 
 type Root = null;
 
+export type ArrayWithPreamble<T> = Array<T> & {_preambleIndex?: number};
```

Comment:
This doesn't make sense. This array is the hotest part so we definitely don't want to do anything that messes with its hidden class. Especially conditionally. I think we need to think about this some more.

Reply:
Yeah i was going to refactor and pass some kind of mutable object in lieu of the preamble but wanted to check with you on the overall approach of pre-flushing up to the first non-preamble chunks. If you are good with that technique I'll just make the API pass a ref-like object that the push function can mutate

Reply:
as discussed we're going back to request level preamble and head open tag always goes in it. better position it carefully